apis/policies/v1/clusteradmissionpolicy_types.go

```diff
@@ -151,7 +151,24 @@ func (r *ClusterAdmissionPolicy) GetRules() []admissionregistrationv1.RuleWithOp
 	return r.Spec.Rules
 }
 
-func (r *ClusterAdmissionPolicy) GetNamespaceSelector() *metav1.LabelSelector {
+func (r *ClusterAdmissionPolicy) GetNamespaceSelector(deploymentNamespace string) *metav1.LabelSelector {
```

Comment:
Since we are changing the namespace selector, 

```suggestion
func (r *ClusterAdmissionPolicy) ReconcileNamespaceSelector(deploymentNamespace string) *metav1.LabelSelector {
```

Or something similar. Granted, that this function is only called in `ReconcileMutatingWebhookConfiguration()` and `UpdateMutatingWebhook()`, but..

Reply:
It doesn't reconcile the namespace selectore inside the method, so I wouldn't add it to the name. This function belongs to the Policy interface, and it will also be called from the [background audit check](https://github.com/kubewarden/audit-scanner/blob/main/internal/policies/fetcher.go#L104) and maybe more places in the future.

Reply:
how about `GetAndUpdateNamespaceSelector`? I'm bound to forget that a getter actually adds a NamespaceSelector and bumps the object.

Feel free to merge as it is, if you still think is overkill.

Reply:
I doesn't really update the selector, just returns a modified version of it. The implementation is different from [ClusterAdmissionPolicy](https://github.com/raulcabello/kubewarden-controller/blob/controller-bug/apis/policies/v1/clusteradmissionpolicy_types.go#L154) and [AdmissionPolicy](https://github.com/kubewarden/kubewarden-controller/blob/main/apis/policies/v1/admissionpolicy_types.go#L118).
What about `GetUpdatedNamespaceSelector`? or `GetModifiedNamespaceSelector`?

Reply:
both are fine! `GetUpdatedNamespaceSelector` sounds good, thanks!

Reply:
changed https://github.com/kubewarden/kubewarden-controller/pull/338/commits/5404bf4333d15f7133b199806aa1a847afb6f1fa